src/mokernels/moinput.jl

```diff
@@ -18,40 +16,63 @@ julia> MOInput(x, 2)
  (3, 2)
 ```
 
-As shown above, an `MOInput` represents a vector of tuples.
+An abstract type to accomodate modelling multi-dimensional output data. There are two
+subtypes that each specify a unique ordering of the dimensions.
+
+As shown above, an `IsotopicByFeatures` represents a vector of tuples.
 The first `length(x)` elements represent the inputs for the first output, the second
 `length(x)` elements represent the inputs for the second output, etc.
 
 See [Inputs for Multiple Outputs](@ref) in the docs for more info.
 """
-struct MOInput{T<:AbstractVector} <: AbstractVector{Tuple{Any,Int}}
+
+struct IsotopicByFeatures{S,T<:AbstractVector{S}} <: AbstractVector{Tuple{T,Int}}
```

Comment:
This should be
```suggestion
struct IsotopicByFeatures{S,T<:AbstractVector{S}} <: AbstractVector{Tuple{S,Int}}
```
shouldn't it?

Reply:
I think you're right. `S` is the type of a single observation and `T` is the vector of those. Re your related [comment](https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/pull/310#discussion_r659760809), I will add a test!